### DISCUSSION TITLE
Google's new model:  recurrent-Gemma (Griffin architecture) - faster inference and lower memory usage when inferencing over long contexts

### DISCUSSION THREAD

**Discussion by joseph777111** [ggml-org/llama.cpp#6605] [General]
@ggerganov Google just dropped a new model architecture: Griffin, which outperforms transformers. And, it offers efficiency advantages including faster inference and lower memory usage when inferencing over longer contexts. [google/gemma.cpp](https://github.com/google/gemma.cpp) has implemented a C++ version of the model (for use and reference). 

Paper: [Google’s Griffin: Mixing Gated Linear Recurrences with Local Attention for Efficient Language Models](https://arxiv.org/pdf/2402.19427.pdf)

[recurrentgemma-2b-it](https://huggingface.co/google/recurrentgemma-2b-it)

**Comment by phymbert** [comment]
Already requested here:
- #6564
